File: apps/java/design-pattern/src/main/java/pattern/adapter/BritishTextObjectAdapter.java
package pattern.adapter;

/**
 * BritishTextObject を TextObject インターフェースに適合させるアダプター。
 * mm をインチに変換し、colour を color として公開する。
 */
public class BritishTextObjectAdapter extends TextObject {
    private static final double MM_PER_INCH = 25.4;
    private final BritishTextObject bto;

    public BritishTextObjectAdapter(BritishTextObject bto) {
        super(null, 0, null);
        this.bto = bto;
    }

    @Override
    public String getText() {
        return bto.getString();
    }

    @Override
    public double getSizeInches() {
        return bto.getSizeMm() / MM_PER_INCH;
    }

    @Override
    public String getColor() {
        return bto.getColour();
    }
}

File: apps/ruby/design-pattern/10-adapter/lib/british_text_object_adapter.rb
require_relative "text_object"
require_relative "british_text_object"

class BritishTextObjectAdapter < TextObject
  def initialize(british_text_object)
    @british_text_object = british_text_object
  end

  def text
    @british_text_object.string
  end

  def size_inches
    @british_text_object.size_mm / 25.4
  end

  def color
    @british_text_object.colour
  end
end
